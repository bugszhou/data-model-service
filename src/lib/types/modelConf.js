/*
 * 数据模型配置参数
 */
/**
 * 基本类型：number string integer boolean object array null
 */

// const demo = {
//   name: '222',
//   age: 12,
//   friends: ['1', '2', '3'],
//   grade: [{ math: 123 }, { math: 123 }, { math: 123 }],
//   family: {
//     dad: '123',
//     mom: '456'
//   },
// };
//
// const model = {
//   name: {
//     type: 'String',
//     description: '姓名',
//     from: 'name',
//     required: false,
//   },
//   friend1: {
//     type: 'String',
//     description: '好友1',
//     from: 'friends[0]',
//     required: false,
//   },
//   grade: {
//     type: 'array',
//     description: '成绩列表',
//     from: 'friends[0]',
//     required: false,
//     properties: gradeModel,
//   },
//   family: {
//     type: 'array',
//     description: '成绩列表',
//     from: 'family',
//     required: false,
//     properties: familyModel,
//   },
// };

// 可用类型
export const typeEnum = ['string', 'number', 'integer', 'boolean', 'object', 'array', 'null'];

export default {
  type: '', // 必填
  description: '',
  from: '', // 必填
  properties: 'none', // 默认none
  default: '', // 默认值，默认为空字符串
  required: false, // 默认false
};
