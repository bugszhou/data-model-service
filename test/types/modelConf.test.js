import modelConf, { typeEnum } from '../../src/lib/types/modelConf';

it('测试 model字段列表', function() {
  expect(modelConf).toMatchSnapshot();
});

it('测试 model的type字段类型集合', () => {
  expect(typeEnum).toMatchSnapshot();
});

