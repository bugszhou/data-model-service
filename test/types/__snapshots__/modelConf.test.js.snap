// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 model字段列表 1`] = `
Object {
  "default": "",
  "description": "",
  "from": "",
  "properties": "none",
  "required": false,
  "type": "",
}
`;

exports[`测试 model的type字段类型集合 1`] = `
Array [
  "string",
  "number",
  "integer",
  "boolean",
  "object",
  "array",
  "null",
]
`;
